fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -1120,24 +1166,30 @@ private boolean needScheduleByShuffleJoin(int fragmentId, DataSink sink) {
     }
 
     private TNetworkAddress toRpcHost(TNetworkAddress host) throws Exception {
-        Backend backend = GlobalStateMgr.getCurrentSystemInfo().getBackendWithBePort(
+        ComputeNode computeNode = GlobalStateMgr.getCurrentSystemInfo().getBackendWithBePort(
                 host.getHostname(), host.getPort());
-        if (backend == null) {
-            throw new UserException("Backend not found. Check if any backend is down or not");
+        if (computeNode == null) {
+            computeNode = GlobalStateMgr.getCurrentSystemInfo().getComputeNodeWithBePort(host.getHostname(), host.getPort());
```

Comment:
getBackendWithBePort  and getComputeNodeWithBePort should combine to one method.